src/ui/common/src/components/integrations/dialogs/s3Dialog.tsx

```diff
@@ -182,6 +201,11 @@ export const S3Dialog: React.FC<Props> = ({ setDialogConfig }) => {
       {s3Type === S3CredentialType.AccessKey && accessKeyTab}
       {s3Type === S3CredentialType.ConfigFilePath && configPathTab}
       {s3Type === S3CredentialType.ConfigFileContent && configUploadTab}
+
+      <FormControlLabel
+        label="Use this integration for Aqueduct storage."
+        control={<Checkbox checked={useAsStorage === 'true'} onChange={(event) => setUseAsStorage(event.target.checked ? 'true' : 'false')} />}
```

Comment:
is useAsStorage a string? Maybe we can do boolean instead?
Might also be able to do:
```checked={useAsStorage}```

Would need to update onChanged method as well to use a boolean.
```onChange={(event) => setUseAsStorage(event.target.checked ? true : false)}```

Reply:
Unfortunately we can't change this from a string to a boolean because of assumptions that our backend makes about what is stored in integration configs. The backend expects everything to be a string, and if this is a boolean it won't be JSON marshalled properly for the backend.

This is tracked in https://linear.app/aqueducthq/issue/ENG-1585/refactor-auth-pkg-to-allow-integration-credential-fields-to-be-non